code/modules/mining/equipment/wormhole_jaunter.dm

```diff
@@ -36,18 +37,29 @@
 
 	return destinations
 
-/obj/item/wormhole_jaunter/proc/activate(mob/user, adjacent)
+/obj/item/wormhole_jaunter/proc/can_jaunter_teleport()
+	var/list/destinations = get_destinations()
+	return destinations.len > 0
+
+/obj/item/wormhole_jaunter/proc/activate(mob/user, adjacent, teleport)
 	if(!turf_check(user))
 		return
 
-	var/list/L = get_destinations(user)
-	if(!L.len)
-		to_chat(user, span_notice("The [src.name] found no beacons in the world to anchor a wormhole to."))
+	var/list/destinations = get_destinations()
+	if(!destinations.len)
```

Comment:
I don't like checking this twice, it smells.
Do you think you could do the same thing you did with chasm react? if it succeeds make activate() return true, otherwise false? that way we're not making duplicate checks

Reply:
Refactored it to return a boolean.  Retested everything and it's good to go.